single_kernel_postgresql/utils/postgresql.py

```diff
@@ -1074,9 +1076,22 @@ def set_up_database(self, temp_location: Optional[str] = None) -> None:
 
             if temp_location is not None:
                 # Fix permissions on the temporary tablespace location when a reboot happens and tmpfs is being used.
-                change_owner(temp_location)
-                os.chmod(temp_location, POSTGRESQL_STORAGE_PERMISSIONS)
+                temp_location_stats = os.stat(temp_location)
+                if (
+                    pwd.getpwuid(temp_location_stats.st_uid).pw_name != SNAP_USER
+                    or temp_location_stats.st_mode != POSTGRESQL_STORAGE_PERMISSIONS
+                ):
+                    change_owner(temp_location)
+                    os.chmod(temp_location, POSTGRESQL_STORAGE_PERMISSIONS)
+                    # Rename existing temp tablespace if it exists, instead of dropping it.
+                    cursor.execute("SELECT TRUE FROM pg_tablespace WHERE spcname='temp';")
+                    if cursor.fetchone() is not None:
+                        new_name = (
+                            f"temp_{datetime.now(timezone.utc).strftime('%Y_%m_%d_%H_%M_%S')}"
+                        )
+                        cursor.execute(f"ALTER TABLESPACE temp RENAME TO {new_name};")
 
```

Comment:
Can we log INFO with count (and names?) of already available (garbage) tablespaces in DB, to hint users for cleanup?